src/Kernel32.Desktop/Kernel32.cs

```diff
@@ -1815,6 +2348,199 @@ public static partial class Kernel32
+
+        [DllImport(nameof(Kernel32), CharSet = CharSet.Unicode, SetLastError = true)]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        public static extern bool AddConsoleAlias(string Source, string Target, string ExeName);
+
+        [DllImport(nameof(Kernel32), SetLastError = true)]
+        public static extern IntPtr CreateConsoleScreenBuffer(uint dwDesiredAccess, uint dwShareMode, ref SECURITY_ATTRIBUTES lpSecurityAttributes, uint dwFlags, IntPtr lpScreenBufferData);
+
+        [DllImport(nameof(Kernel32), SetLastError = true)]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        public static extern bool AttachConsole(uint dwProcessId);
+
+        [DllImport(nameof(Kernel32), SetLastError = true)]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        public static extern bool FlushConsoleInputBuffer(IntPtr hConsoleInput);
+
+        [DllImport(nameof(Kernel32), CharSet = CharSet.Unicode, SetLastError = true)]
+        public static extern unsafe uint GetConsoleAliases(
+            [Friendly(FriendlyFlags.Out | FriendlyFlags.Array)] char* lpAliasBuffer,
+            int AliasBufferLength,
+            string lpExeName);
+
+        [DllImport(nameof(Kernel32), CharSet = CharSet.Unicode, SetLastError = true)]
+        public static extern uint GetConsoleAliasesLength(string lpExeName);
+
+        [DllImport(nameof(Kernel32), CharSet = CharSet.Unicode, SetLastError = true)]
+        public static extern unsafe uint GetConsoleAliasExes(
+            [Friendly(FriendlyFlags.Out | FriendlyFlags.Array)] char* lpExeNameBuffer,
+            uint ExeNameBufferLength);
+
+        [DllImport(nameof(Kernel32), SetLastError = true)]
+        public static extern uint GetConsoleAliasExesLength();
+
+        [DllImport(nameof(Kernel32), CharSet = CharSet.Unicode, SetLastError = true)]
+        public static extern unsafe uint GetConsoleAlias(
+            string lpSource,
+            [Friendly(FriendlyFlags.Out | FriendlyFlags.Array)] char* lpTargetBuffer,
+            uint TargetBufferLength,
+            string lpExeName);
+
+        [DllImport(nameof(Kernel32), SetLastError = true)]
+        public static extern uint GetConsoleCP();
+
+        [DllImport(nameof(Kernel32), SetLastError = true)]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        public static extern bool GetConsoleDisplayMode(ref uint lpModeFlags);
+
+        [DllImport(nameof(Kernel32), SetLastError = true)]
+        public static extern COORD GetConsoleFontSize(IntPtr hConsoleOutput, uint nFont);
+
+        [DllImport(nameof(Kernel32), SetLastError = true)]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        public static extern bool GetConsoleMode(IntPtr hConsoleHandle, ref uint lpMode);
+
+        [DllImport("kernel32.dll", EntryPoint = "SetConsoleMode")]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        public static extern bool SetConsoleMode(IntPtr hConsoleHandle, uint dwMode);
```

Comment:
You've really done a great deal in this PR. I really appreciate it. Any interest in defining an enum for the values that `dwMode` can be?